app/src/main/kotlin/it/vfsfitvnm/vimusic/service/PlayerService.kt

```diff
@@ -1099,10 +1099,13 @@ class PlayerService : InvincibleService(), Player.Listener, PlaybackStatsListene
             chunkLength: Long? = DEFAULT_CHUNK_LENGTH
         ): ResolvingDataSource.Factory {
             val ringBuffer = RingBuffer<Pair<String, Uri>?>(2) { null }
+            val cacheDataSource = CacheDataSource.Factory().setCache(cache)
+            if (!PlayerPreferences.progressiveCache)
```

Comment:
What do you mean by progressive cache? You just made the cache readonly here. What do you think this solves for the user? I think the best way to handle this is to just add another option for the max cache size: 0 bytes.

Reply:
Disabling 'Progressive cache' is what i called when user just want to play the song without caching or storing the song in disk. Keeping it in memory. (This could be useful for users that just want to )
Max cache size as 0 it's also a good idea, but on original fork i saw the user can store as many songs in disk in one session, and once he leave or restart the app, it starts freeing the usage disk.

Reply:
Well, how it works right now is not consistent behavior, the setting only gets applied after a full restart of the app. I think we should indeed just add a new max cache size.